client/components/NewStory.tsx

```diff
@@ -55,6 +56,10 @@ export default function NewStory() {
       return;
     }
 
+    const image_url = await createImage(readingLevelOptions[data.readingLevel[0]], data.location[0], data.themes);
+    const filename = `${storyDetails.id}.jpeg`
+    await storeStoryImage(image_url, filename);
```

Comment:
this needs a bit of work but we can worry about it later.

we should not wait for the image to be generated/stored before we show the user the story

Reply:
Also you are waiting for the story to be generated before you request the image. you could use Promis.all something like this:
```
async function fetchMultipleAPIs(urls) {
  try {
    const fetchPromises = urls.map(url => fetch(url));
    const responses = await Promise.all(fetchPromises);

    const jsonPromises = responses.map(response => response.json());
    const results = await Promise.all(jsonPromises);

    return results;
  } catch (error) {
    console.error('Error fetching APIs:', error);
    throw error;
  }
}

// Example usage:
const apiUrls = ['https://api.example.com/data1', 'https://api.example.com/data2'];

fetchMultipleAPIs(apiUrls)
  .then(results => {
    console.log('API results:', results);
  })
  .catch(error => {
    console.error('Error:', error);
  });
  ```